File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        int needleLength = needle.size();
        int haystackLength = haystack.size();
        for(int i = 0;i<=haystackLength-needleLength;i++){
            string sub = haystack.substr(i,needleLength);
            if(sub == needle){
                return i;
            }
        }
        return -1;
    }
};

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.java
class Solution {
    public int strStr(String haystack, String needle) {
        int needleLength = needle.length();
        int haystackLength = haystack.length();
        for(int i =0;i<=haystackLength-needleLength;i++){
            if(i==haystackLength-needleLength){
                String haysub = haystack.substring(i);
                if(haysub.equals(needle)){
                    return i;
                }
            }else{
                String haysub = haystack.substring(i,i+needleLength);
                if(haysub.equals(needle)){
                    return i;
                }
            }
        }
        return -1;
    }
}